Source/JavaScriptCore/b3/air/AirCCallSpecial.cpp

```diff
@@ -51,13 +51,13 @@ void CCallSpecial::forEachArg(Inst& inst, const ScopedLambda<Inst::EachArgCallba
     for (unsigned i = 0; i < numReturnGPArgs; ++i)
         callback(inst.args[returnGPArgOffset + i], Arg::Def, GP, pointerWidth());
     for (unsigned i = 0; i < numReturnFPArgs; ++i)
-        callback(inst.args[returnFPArgOffset + i], Arg::Def, FP, Width64);
+        callback(inst.args[returnFPArgOffset + i], Arg::Def, FP, Options::useWebAssemblySIMD() ? conservativeWidth(FP) : conservativeWidthWithoutVectors(FP));
```

Comment:
Once `useWebAssemblySIMD` is enabled, it starts having conservative width (128bit), this sounds regression from JS side. Probably we need a flag in `CCallSpecial` which suppresses this behavior when it is used in pure JS context. Can you add FIXME about it here?

Reply:
I actually think that the register allocator should be able to ignore saving the top 64 bits if nobody USEs them, once the graph coloring and linear scan allocators support vectors. Thoughts?

Reply:
That sounds good to me. Anyway, either is fine if we do not regress the pure JS code generation. Please do either of them before enabling SIMD, otherwise, it gets regressed.